<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Create signature matrix using CIBERSORTx — create_signature_matrix • proteoDeconv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Create signature matrix using CIBERSORTx — create_signature_matrix"><meta name="description" content="Generates a signature matrix from reference profiles and cell type classes
using the CIBERSORTx Docker image.
This signature matrix contains cell type-specific marker proteins that will be used for deconvolution."><meta property="og:description" content="Generates a signature matrix from reference profiles and cell type classes
using the CIBERSORTx Docker image.
This signature matrix contains cell type-specific marker proteins that will be used for deconvolution."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">proteoDeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/proteoDeconv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ComputationalProteomics/proteoDeconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Create signature matrix using CIBERSORTx</h1>
      <small class="dont-index">Source: <a href="https://github.com/ComputationalProteomics/proteoDeconv/blob/main/R/create_signature_matrix.R" class="external-link"><code>R/create_signature_matrix.R</code></a></small>
      <div class="d-none name"><code>create_signature_matrix.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Generates a signature matrix from reference profiles and cell type classes
using the CIBERSORTx Docker image.
This signature matrix contains cell type-specific marker proteins that will be used for deconvolution.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">create_signature_matrix</span><span class="op">(</span></span>
<span>  refsample <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  phenoclasses <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  g_min <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  g_max <span class="op">=</span> <span class="fl">400</span>,</span>
<span>  q_value <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>  replicates <span class="op">=</span> <span class="fl">5</span>,</span>
<span>  sampling <span class="op">=</span> <span class="fl">0.5</span>,</span>
<span>  fraction <span class="op">=</span> <span class="fl">0.75</span>,</span>
<span>  filter <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  QN <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  single_cell <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  use_sudo <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  <span class="va">...</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-refsample">refsample<a class="anchor" aria-label="anchor" href="#arg-refsample"></a></dt>
<dd><p>A numeric matrix containing reference profiles with genes as row names and samples as columns.
This should be preprocessed data from pure cell populations.</p></dd>


<dt id="arg-phenoclasses">phenoclasses<a class="anchor" aria-label="anchor" href="#arg-phenoclasses"></a></dt>
<dd><p>A numeric matrix, data frame, or tibble containing cell type classification (0/1/2).
If provided as a matrix or base data frame, the cell types are assumed to be in the row names.
If provided as a tibble, the cell type identifiers should be included as a column.
This is typically created using the create_phenoclasses() function.</p></dd>


<dt id="arg-g-min">g_min<a class="anchor" aria-label="anchor" href="#arg-g-min"></a></dt>
<dd><p>Minimum number of genes per cell type in the signature matrix. Default is 200.</p></dd>


<dt id="arg-g-max">g_max<a class="anchor" aria-label="anchor" href="#arg-g-max"></a></dt>
<dd><p>Maximum number of genes per cell type in the signature matrix. Default is 400.</p></dd>


<dt id="arg-q-value">q_value<a class="anchor" aria-label="anchor" href="#arg-q-value"></a></dt>
<dd><p>Q-value threshold for differential expression. Default is 0.01.</p></dd>


<dt id="arg-replicates">replicates<a class="anchor" aria-label="anchor" href="#arg-replicates"></a></dt>
<dd><p>Number of replicates to use for building the reference file (only relevant when single_cell=TRUE). Default is 5.</p></dd>


<dt id="arg-sampling">sampling<a class="anchor" aria-label="anchor" href="#arg-sampling"></a></dt>
<dd><p>Fraction of available gene expression profiles selected by random sampling (only relevant when single_cell=TRUE). Default is 0.5.</p></dd>


<dt id="arg-fraction">fraction<a class="anchor" aria-label="anchor" href="#arg-fraction"></a></dt>
<dd><p>Fraction of cells of the same identity showing evidence of expression (only relevant when single_cell=TRUE). Default is 0.75.</p></dd>


<dt id="arg-filter">filter<a class="anchor" aria-label="anchor" href="#arg-filter"></a></dt>
<dd><p>Logical indicating whether to remove non-hematopoietic genes. Default is FALSE.</p></dd>


<dt id="arg-verbose">verbose<a class="anchor" aria-label="anchor" href="#arg-verbose"></a></dt>
<dd><p>Logical indicating whether to print detailed output. Default is FALSE.</p></dd>


<dt id="arg-qn">QN<a class="anchor" aria-label="anchor" href="#arg-qn"></a></dt>
<dd><p>Logical indicating whether to run quantile normalization. Default is FALSE.</p></dd>


<dt id="arg-single-cell">single_cell<a class="anchor" aria-label="anchor" href="#arg-single-cell"></a></dt>
<dd><p>Logical indicating whether to create signature from scRNA-Seq data. Default is FALSE.</p></dd>


<dt id="arg-use-sudo">use_sudo<a class="anchor" aria-label="anchor" href="#arg-use-sudo"></a></dt>
<dd><p>Logical indicating whether to use sudo for Docker commands. Default is FALSE.</p></dd>


<dt id="arg--">...<a class="anchor" aria-label="anchor" href="#arg--"></a></dt>
<dd><p>Additional arguments passed to the function.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A numeric matrix with genes as rows and cell types as columns, representing
the expression profile of each cell type. This signature matrix can be used
as input for deconvolution algorithms to estimate cell type proportions
in mixed samples.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function uses the CIBERSORTx Docker image to construct a signature matrix
based on the input reference profiles and cell type classifications. The CIBERSORTx token
and email must be set as environment variables either in the project directory's .Renviron
file or in the user's home directory .Renviron file.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="create_phenoclasses.html">create_phenoclasses</a></code> for creating the phenoclasses input and
<code><a href="deconvolute.html">deconvolute</a></code> for using the signature matrix in deconvolution.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Load preprocessed pure samples data</span></span></span>
<span class="r-in"><span><span class="va">pure_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pure_samples_matrix.rds"</span>,</span></span>
<span class="r-in"><span>                                    package <span class="op">=</span> <span class="st">"proteoDeconv"</span><span class="op">)</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create phenoclasses for the samples</span></span></span>
<span class="r-in"><span><span class="va">mapping_rules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  <span class="st">"CD8+ T cells"</span> <span class="op">=</span> <span class="st">"CD8"</span>,</span></span>
<span class="r-in"><span>  <span class="st">"Monocytes"</span> <span class="op">=</span> <span class="st">"Mono"</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">phenoclasses</span> <span class="op">&lt;-</span> <span class="fu"><a href="create_phenoclasses.html">create_phenoclasses</a></span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">pure_samples</span>,</span></span>
<span class="r-in"><span>  mapping_rules <span class="op">=</span> <span class="va">mapping_rules</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Create signature matrix</span></span></span>
<span class="r-in"><span><span class="va">signature_matrix</span> <span class="op">&lt;-</span> <span class="fu">create_signature_matrix</span><span class="op">(</span></span></span>
<span class="r-in"><span>  refsample <span class="op">=</span> <span class="va">pure_samples</span>,</span></span>
<span class="r-in"><span>  phenoclasses <span class="op">=</span> <span class="va">phenoclasses</span>,</span></span>
<span class="r-in"><span>  g_min <span class="op">=</span> <span class="fl">200</span>,</span></span>
<span class="r-in"><span>  g_max <span class="op">=</span> <span class="fl">400</span>,</span></span>
<span class="r-in"><span>  q_value <span class="op">=</span> <span class="fl">0.01</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Måns Zamore.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.1.</p>
</div>

    </footer></div>





  </body></html>

