<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Deconvolute using CIBERSORTx Docker image — deconvolute_cibersortx • proteoDeconv</title><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet"><link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet"><script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Deconvolute using CIBERSORTx Docker image — deconvolute_cibersortx"><meta name="description" content="Performs deconvolution of bulk proteome data into constituent cell types
using the CIBERSORTx Docker image. This function handles the interaction with
the Docker container and processes the results."><meta property="og:description" content="Performs deconvolution of bulk proteome data into constituent cell types
using the CIBERSORTx Docker image. This function handles the interaction with
the Docker container and processes the results."></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">proteoDeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0.9000</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item"><a class="nav-link" href="../articles/proteoDeconv.html">Get started</a></li>
<li class="active nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul><ul class="navbar-nav"><li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json"></form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ComputationalProteomics/proteoDeconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul></div>


  </div>
</nav><div class="container template-reference-topic">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Deconvolute using CIBERSORTx Docker image</h1>
      <small class="dont-index">Source: <a href="https://github.com/ComputationalProteomics/proteoDeconv/blob/main/R/deconvolute_cibersortx.R" class="external-link"><code>R/deconvolute_cibersortx.R</code></a></small>
      <div class="d-none name"><code>deconvolute_cibersortx.Rd</code></div>
    </div>

    <div class="ref-description section level2">
    <p>Performs deconvolution of bulk proteome data into constituent cell types
using the CIBERSORTx Docker image. This function handles the interaction with
the Docker container and processes the results.</p>
    </div>

    <div class="section level2">
    <h2 id="ref-usage">Usage<a class="anchor" aria-label="anchor" href="#ref-usage"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span><span class="fu">deconvolute_cibersortx</span><span class="op">(</span></span>
<span>  <span class="va">data</span>,</span>
<span>  <span class="va">signature</span>,</span>
<span>  perm <span class="op">=</span> <span class="fl">1</span>,</span>
<span>  rmbatch_S_mode <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  source_GEPs <span class="op">=</span> <span class="cn">NULL</span>,</span>
<span>  rmbatch_B_mode <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  QN <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  absolute <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>  abs_method <span class="op">=</span> <span class="st">"sig.score"</span>,</span>
<span>  use_sudo <span class="op">=</span> <span class="cn">FALSE</span></span>
<span><span class="op">)</span></span></code></pre></div>
    </div>

    <div class="section level2">
    <h2 id="arguments">Arguments<a class="anchor" aria-label="anchor" href="#arguments"></a></h2>


<dl><dt id="arg-data">data<a class="anchor" aria-label="anchor" href="#arg-data"></a></dt>
<dd><p>A numeric matrix containing mixture data with genes as row names
and samples as columns.</p></dd>


<dt id="arg-signature">signature<a class="anchor" aria-label="anchor" href="#arg-signature"></a></dt>
<dd><p>A numeric matrix containing the signature matrix with genes
as row names and cell types as columns.</p></dd>


<dt id="arg-perm">perm<a class="anchor" aria-label="anchor" href="#arg-perm"></a></dt>
<dd><p>An integer specifying the number of permutations to be performed.
Default is 1.</p></dd>


<dt id="arg-rmbatch-s-mode">rmbatch_S_mode<a class="anchor" aria-label="anchor" href="#arg-rmbatch-s-mode"></a></dt>
<dd><p>A logical value indicating whether to remove batch
effects in source GEPs mode. Default is FALSE.</p></dd>


<dt id="arg-source-geps">source_GEPs<a class="anchor" aria-label="anchor" href="#arg-source-geps"></a></dt>
<dd><p>A matrix containing the source gene expression profiles.
Required if <code>rmbatch_S_mode</code> is TRUE.</p></dd>


<dt id="arg-rmbatch-b-mode">rmbatch_B_mode<a class="anchor" aria-label="anchor" href="#arg-rmbatch-b-mode"></a></dt>
<dd><p>A logical value indicating whether to remove batch
effects in bulk mode. Default is FALSE.</p></dd>


<dt id="arg-qn">QN<a class="anchor" aria-label="anchor" href="#arg-qn"></a></dt>
<dd><p>A logical value indicating whether to perform quantile
normalization. Default is FALSE.</p></dd>


<dt id="arg-absolute">absolute<a class="anchor" aria-label="anchor" href="#arg-absolute"></a></dt>
<dd><p>A logical value indicating whether to use absolute mode.
Default is FALSE.</p></dd>


<dt id="arg-abs-method">abs_method<a class="anchor" aria-label="anchor" href="#arg-abs-method"></a></dt>
<dd><p>A character string specifying the method to use for
absolute mode. Default is "sig.score".</p></dd>


<dt id="arg-use-sudo">use_sudo<a class="anchor" aria-label="anchor" href="#arg-use-sudo"></a></dt>
<dd><p>A logical value indicating whether to use sudo for Docker
commands. Default is FALSE.</p></dd>

</dl></div>
    <div class="section level2">
    <h2 id="value">Value<a class="anchor" aria-label="anchor" href="#value"></a></h2>
    <p>A numeric matrix with samples as rows and cell types as columns,
representing the estimated proportion of each cell type in each sample.</p>
    </div>
    <div class="section level2">
    <h2 id="details">Details<a class="anchor" aria-label="anchor" href="#details"></a></h2>
    <p>This function requires the CIBERSORTx Docker image to be installed
and the <code>CIBERSORTX_EMAIL</code> and <code>CIBERSORTX_TOKEN</code> environment
variables to be set. You can get these credentials by registering at the
CIBERSORTx website (https://cibersortx.stanford.edu/).</p>
<p>The function creates temporary files for the mixture data and signature
matrix, runs the CIBERSORTx Docker container, and processes the results. Note
that absolute mode is not currently supported in the Docker version.</p>
    </div>
    <div class="section level2">
    <h2 id="see-also">See also<a class="anchor" aria-label="anchor" href="#see-also"></a></h2>
    <div class="dont-index"><p><code><a href="deconvolute_cibersort.html">deconvolute_cibersort</a></code> for using the R implementation
of CIBERSORT, <code><a href="deconvolute.html">deconvolute</a></code> for a unified interface to multiple
deconvolution methods.</p></div>
    </div>

    <div class="section level2">
    <h2 id="ref-examples">Examples<a class="anchor" aria-label="anchor" href="#ref-examples"></a></h2>
    <div class="sourceCode"><pre class="sourceCode r"><code><span class="r-in"><span><span class="kw">if</span> <span class="op">(</span><span class="cn">FALSE</span><span class="op">)</span> <span class="op">{</span> <span class="co"># \dontrun{</span></span></span>
<span class="r-in"><span><span class="co"># Set required environment variables (ideally in .Renviron)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>CIBERSORTX_EMAIL <span class="op">=</span> <span class="st">"your.email@example.com"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="fu"><a href="https://rdrr.io/r/base/Sys.setenv.html" class="external-link">Sys.setenv</a></span><span class="op">(</span>CIBERSORTX_TOKEN <span class="op">=</span> <span class="st">"your-token-here"</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Load example data and signature matrix</span></span></span>
<span class="r-in"><span><span class="va">data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mixed_samples_matrix.rds"</span>, package <span class="op">=</span> <span class="st">"proteoDeconv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">mixed_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">data_file</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="va">signature_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"cd8t_mono_signature_matrix.rds"</span>, package <span class="op">=</span> <span class="st">"proteoDeconv"</span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="va">signature_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">signature_file</span><span class="op">)</span></span></span>
<span class="r-in"><span></span></span>
<span class="r-in"><span><span class="co"># Run deconvolution with CIBERSORTx Docker</span></span></span>
<span class="r-in"><span><span class="va">result</span> <span class="op">&lt;-</span> <span class="fu">deconvolute_cibersortx</span><span class="op">(</span></span></span>
<span class="r-in"><span>  data <span class="op">=</span> <span class="va">mixed_samples</span>,</span></span>
<span class="r-in"><span>  signature <span class="op">=</span> <span class="va">signature_matrix</span>,</span></span>
<span class="r-in"><span>  perm <span class="op">=</span> <span class="fl">100</span>,</span></span>
<span class="r-in"><span>  QN <span class="op">=</span> <span class="cn">TRUE</span></span></span>
<span class="r-in"><span><span class="op">)</span></span></span>
<span class="r-in"><span><span class="op">}</span> <span class="co"># }</span></span></span>
<span class="r-in"><span></span></span>
</code></pre></div>
    </div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p>Developed by Måns Zamore.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer></div>





  </body></html>

