<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Introduction to proteoDeconv • proteoDeconv</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.5.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.5.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Introduction to proteoDeconv">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-primary" data-bs-theme="dark" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">proteoDeconv</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="active nav-item"><a class="nav-link" href="../articles/proteoDeconv.html">Get started</a></li>
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/ComputationalProteomics/proteoDeconv/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Introduction to proteoDeconv</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/ComputationalProteomics/proteoDeconv/blob/main/vignettes/proteoDeconv.Rmd" class="external-link"><code>vignettes/proteoDeconv.Rmd</code></a></small>
      <div class="d-none name"><code>proteoDeconv.Rmd</code></div>
    </div>

    
    
<div class="section level2">
<h2 id="introduction">Introduction<a class="anchor" aria-label="anchor" href="#introduction"></a>
</h2>
<p>This vignette describes a basic workflow that can be used to run the
proteoDeconv package.</p>
</div>
<div class="section level2">
<h2 id="loading-required-packages">Loading required packages<a class="anchor" aria-label="anchor" href="#loading-required-packages"></a>
</h2>
<p>First, let’s load the necessary packages:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://github.com/ComputationalProteomics/proteoDeconv" class="external-link">proteoDeconv</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://readr.tidyverse.org" class="external-link">readr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="example-data-overview">Example data overview<a class="anchor" aria-label="anchor" href="#example-data-overview"></a>
</h2>
<p><code>proteoDeconv</code> includes two example datasets, which are
subsets of our dataset PXD056050:</p>
<ol style="list-style-type: decimal">
<li>Pure cell type samples: Proteomic measurements from isolated CD8+ T
cells and monocytes.</li>
<li>Mixed samples: Proteomic data from experimental mixtures containing
50% CD8+ T cells and 50% monocytes.</li>
</ol>
<p>Let’s load these datasets:</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pure_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"pure_samples_matrix.rds"</span>, package <span class="op">=</span> <span class="st">"proteoDeconv"</span><span class="op">)</span></span>
<span><span class="va">pure_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">pure_data_file</span><span class="op">)</span></span>
<span></span>
<span><span class="va">mix_data_file</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/system.file.html" class="external-link">system.file</a></span><span class="op">(</span><span class="st">"extdata"</span>, <span class="st">"mixed_samples_matrix.rds"</span>, package <span class="op">=</span> <span class="st">"proteoDeconv"</span><span class="op">)</span></span>
<span><span class="va">mixed_samples</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/readRDS.html" class="external-link">readRDS</a></span><span class="op">(</span><span class="va">mix_data_file</span><span class="op">)</span></span></code></pre></div>
<p>Let’s examine the structure of the pure samples data:</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">pure_samples</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 3317    8</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pure_samples</span><span class="op">)</span></span>
<span><span class="co">#&gt;                            CD8+ T cells_1 CD8+ T cells_2 CD8+ T cells_3</span></span>
<span><span class="co">#&gt; IGKV3-15;IGKV3-7;IGKV3D-7              NA         799607       158030.0</span></span>
<span><span class="co">#&gt; IGLV3-16;IGLV3-25;IGLV3-27             NA             NA             NA</span></span>
<span><span class="co">#&gt; IGKV3-11;IGKV3D-11                     NA         230449        41679.4</span></span>
<span><span class="co">#&gt; IGHV3OR16-9                            NA         151730        41122.8</span></span>
<span><span class="co">#&gt; GATD3;GATD3B                      2297760        2768020      3160440.0</span></span>
<span><span class="co">#&gt; IGKV1-13;IGKV1D-13                     NA             NA             NA</span></span>
<span><span class="co">#&gt;                            CD8+ T cells_5 Monocytes_1 Monocytes_2 Monocytes_3</span></span>
<span><span class="co">#&gt; IGKV3-15;IGKV3-7;IGKV3D-7          477644      213180          NA      149232</span></span>
<span><span class="co">#&gt; IGLV3-16;IGLV3-25;IGLV3-27             NA          NA          NA          NA</span></span>
<span><span class="co">#&gt; IGKV3-11;IGKV3D-11                 150046          NA          NA          NA</span></span>
<span><span class="co">#&gt; IGHV3OR16-9                            NA          NA          NA          NA</span></span>
<span><span class="co">#&gt; GATD3;GATD3B                      2765990     1864430     1859990     1653820</span></span>
<span><span class="co">#&gt; IGKV1-13;IGKV1D-13                     NA          NA          NA          NA</span></span>
<span><span class="co">#&gt;                            Monocytes_4</span></span>
<span><span class="co">#&gt; IGKV3-15;IGKV3-7;IGKV3D-7           NA</span></span>
<span><span class="co">#&gt; IGLV3-16;IGLV3-25;IGLV3-27          NA</span></span>
<span><span class="co">#&gt; IGKV3-11;IGKV3D-11                  NA</span></span>
<span><span class="co">#&gt; IGHV3OR16-9                         NA</span></span>
<span><span class="co">#&gt; GATD3;GATD3B                   1906570</span></span>
<span><span class="co">#&gt; IGKV1-13;IGKV1D-13                  NA</span></span></code></pre></div>
<p>The data matrix contains protein measurements (rows) across different
samples (columns). The sample names indicate the cell type (CD8 for CD8+
T cells and Mono for monocytes).</p>
</div>
<div class="section level2">
<h2 id="data-preprocessing">Data preprocessing<a class="anchor" aria-label="anchor" href="#data-preprocessing"></a>
</h2>
<p>Proper preprocessing is important for accurate deconvolution. The
preprocessing pipeline in <code>proteoDeconv</code> includes several
important steps:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pure_samples_preprocessed</span> <span class="op">&lt;-</span> <span class="va">pure_samples</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Extract first gene group when multiple gene groups are present</span></span>
<span>  <span class="fu"><a href="../reference/extract_identifiers.html">extract_identifiers</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Update gene symbols to current HGNC nomenclature</span></span>
<span>  <span class="fu"><a href="../reference/update_gene_symbols.html">update_gene_symbols</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Handle missing values by imputing with the lowest observed value</span></span>
<span>  <span class="fu"><a href="../reference/handle_missing_values.html">handle_missing_values</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Resolve duplicate genes by keeping the row with highest median value</span></span>
<span>  <span class="fu"><a href="../reference/handle_duplicates.html">handle_duplicates</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="co"># Scale the data (TPM-like normalization)</span></span>
<span>  <span class="fu"><a href="../reference/convert_to_tpm.html">convert_to_tpm</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># Apply the same preprocessing to mixed samples</span></span>
<span><span class="va">mixed_samples_preprocessed</span> <span class="op">&lt;-</span> <span class="va">mixed_samples</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/extract_identifiers.html">extract_identifiers</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/update_gene_symbols.html">update_gene_symbols</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/handle_missing_values.html">handle_missing_values</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/handle_duplicates.html">handle_duplicates</a></span><span class="op">(</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="../reference/convert_to_tpm.html">convert_to_tpm</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p>Let’s examine the preprocessed data:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">pure_samples_preprocessed</span><span class="op">)</span></span>
<span><span class="co">#&gt;             CD8+ T cells_1 CD8+ T cells_2 CD8+ T cells_3 CD8+ T cells_5</span></span>
<span><span class="co">#&gt; IGKV3-15        0.08027906    19.66485501      8.2758047     22.9537904</span></span>
<span><span class="co">#&gt; IGLV3-16        0.08027906     0.06133261      0.1306014      0.1198471</span></span>
<span><span class="co">#&gt; IGKV3-11        0.08027906     5.66746686      2.1826905      7.2106515</span></span>
<span><span class="co">#&gt; IGHV3OR16-9     0.08027906     3.73151867      2.1535421      0.1198471</span></span>
<span><span class="co">#&gt; GATD3          73.96557868    68.07433147    165.5077151    132.9231702</span></span>
<span><span class="co">#&gt; IGKV1-13        0.08027906     0.06133261      0.1306014      0.1198471</span></span>
<span><span class="co">#&gt;             Monocytes_1 Monocytes_2 Monocytes_3 Monocytes_4</span></span>
<span><span class="co">#&gt; IGKV3-15      13.462733   0.1468999   9.8914941   0.1450565</span></span>
<span><span class="co">#&gt; IGLV3-16       0.157494   0.1468999   0.1653017   0.1450565</span></span>
<span><span class="co">#&gt; IGKV3-11       0.157494   0.1468999   0.1653017   0.1450565</span></span>
<span><span class="co">#&gt; IGHV3OR16-9    0.157494   0.1468999   0.1653017   0.1450565</span></span>
<span><span class="co">#&gt; GATD3        117.742395 109.5607297 109.6195913 110.8951904</span></span>
<span><span class="co">#&gt; IGKV1-13       0.157494   0.1468999   0.1653017   0.1450565</span></span></code></pre></div>
<div class="section level3">
<h3 id="preprocessing-options">Preprocessing options<a class="anchor" aria-label="anchor" href="#preprocessing-options"></a>
</h3>
<p><code>proteoDeconv</code> offers several options for each
preprocessing step, which have been evaluated for their impact on
deconvolution performance. For further information on the options,
please refer to the <a href="https://computationalproteomics.github.io/proteoDeconv/reference/index.html">function
reference</a>.</p>
</div>
</div>
<div class="section level2">
<h2 id="creating-a-signature-matrix">Creating a signature matrix<a class="anchor" aria-label="anchor" href="#creating-a-signature-matrix"></a>
</h2>
<p>A signature matrix contains cell type-specific marker proteins that
will be used for the deconvolution. Signature matrices represent the
reference profiles of pure cell populations.</p>
<p>To create one, we first need to establish which samples correspond to
which cell types:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">mapping_rules</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/list.html" class="external-link">list</a></span><span class="op">(</span></span>
<span>  <span class="st">"CD8+ T cells"</span> <span class="op">=</span> <span class="st">"CD8"</span>, <span class="co"># Samples containing "CD8" will be classified as CD8+ T cells</span></span>
<span>  <span class="st">"Monocytes"</span> <span class="op">=</span> <span class="st">"Mono"</span> <span class="co"># Samples containing "Mono" will be classified as Monocytes</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="va">phenoclasses</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_phenoclasses.html">create_phenoclasses</a></span><span class="op">(</span></span>
<span>  <span class="va">pure_samples_preprocessed</span>,</span>
<span>  <span class="va">mapping_rules</span>,</span>
<span>  verbose <span class="op">=</span> <span class="cn">TRUE</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Using regex-based mapping rules</span></span>
<span><span class="co">#&gt; Mapping to 'CD8+ T cells': CD8+ T cells_1, CD8+ T cells_2, CD8+ T cells_3, CD8+ T cells_5</span></span>
<span><span class="co">#&gt; Mapping to 'Monocytes': Monocytes_1, Monocytes_2, Monocytes_3, Monocytes_4</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">phenoclasses</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 2 × 9</span></span></span>
<span><span class="co">#&gt;   cell_type  `CD8+ T cells_1` `CD8+ T cells_2` `CD8+ T cells_3` `CD8+ T cells_5`</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                 <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>            <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> CD8+ T ce…                1                1                1                1</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Monocytes                 2                2                2                2</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4 more variables: Monocytes_1 &lt;dbl&gt;, Monocytes_2 &lt;dbl&gt;, Monocytes_3 &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   Monocytes_4 &lt;dbl&gt;</span></span></span></code></pre></div>
<p>The phenoclasses matrix shows which cell type each sample represents,
which is required for the signature matrix generation.</p>
<p>Now, we can create the signature matrix using CIBERSORTx:</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Note: This requires a CIBERSORTx token to be set in .Renviron</span></span>
<span><span class="va">signature_matrix</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/create_signature_matrix.html">create_signature_matrix</a></span><span class="op">(</span></span>
<span>  refsample <span class="op">=</span> <span class="va">pure_samples_preprocessed</span>,</span>
<span>  phenoclasses <span class="op">=</span> <span class="va">phenoclasses</span>,</span>
<span>  g_min <span class="op">=</span> <span class="fl">200</span>,</span>
<span>  g_max <span class="op">=</span> <span class="fl">400</span>,</span>
<span>  q_value <span class="op">=</span> <span class="fl">0.01</span></span>
<span><span class="op">)</span></span></code></pre></div>
<p>Let’s examine the signature matrix:</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/base/dim.html" class="external-link">dim</a></span><span class="op">(</span><span class="va">signature_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt; [1] 426   2</span></span>
<span></span>
<span><span class="co"># Look at the first rows</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">signature_matrix</span><span class="op">)</span></span>
<span><span class="co">#&gt;        CD8+ T cells  Monocytes</span></span>
<span><span class="co">#&gt; ABCB6    77.6657360 353.951727</span></span>
<span><span class="co">#&gt; ABI3     34.3210650   0.152197</span></span>
<span><span class="co">#&gt; ABLIM1   12.5659965   0.152197</span></span>
<span><span class="co">#&gt; ACAA1     0.1000631   3.507357</span></span>
<span><span class="co">#&gt; ACADVL   55.3678625 222.706106</span></span>
<span><span class="co">#&gt; ACSL3     2.3622472  14.178111</span></span></code></pre></div>
<p>The signature matrix contains proteins (rows) with their expression
values in each cell type (columns). These proteins were selected based
on their ability to differentiate between CD8+ T cells and
monocytes.</p>
</div>
<div class="section level2">
<h2 id="performing-deconvolution">Performing deconvolution<a class="anchor" aria-label="anchor" href="#performing-deconvolution"></a>
</h2>
<p>Now we can deconvolute our mixed samples to estimate their cell type
composition. The <code>proteoDeconv</code> package supports multiple
algorithms that have been evaluated for their performance with
proteomics data. We’ll use the EPIC algorithm here <span class="citation">(Racle et al. 2017)</span>:</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  algorithm <span class="op">=</span> <span class="st">"epic"</span>,</span>
<span>  data <span class="op">=</span> <span class="va">mixed_samples_preprocessed</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature_matrix</span>,</span>
<span>  with_other_cells <span class="op">=</span> <span class="cn">FALSE</span> <span class="co"># No need to calculate other cells in this case</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/base/print.html" class="external-link">print</a></span><span class="op">(</span><span class="va">results</span><span class="op">)</span></span>
<span><span class="co">#&gt;       CD8+ T cells Monocytes</span></span>
<span><span class="co">#&gt; Mix_1    0.2691316 0.7308684</span></span>
<span><span class="co">#&gt; Mix_2    0.3341222 0.6658778</span></span>
<span><span class="co">#&gt; Mix_3    0.2074690 0.7925310</span></span>
<span><span class="co">#&gt; Mix_4    0.4650601 0.5349399</span></span></code></pre></div>
<p>The results show the estimated proportion of each cell type in each
mixed sample.</p>
<div class="section level3">
<h3 id="available-deconvolution-algorithms">Available deconvolution algorithms<a class="anchor" aria-label="anchor" href="#available-deconvolution-algorithms"></a>
</h3>
<p><code>proteoDeconv</code> supports multiple deconvolution
algorithms:</p>
<ul>
<li>
<strong>EPIC</strong> <span class="citation">(Racle et al.
2017)</span>: A reference-based method that uses constrained
least-squares regression</li>
<li>
<strong>CIBERSORT</strong> <span class="citation">(Newman et al.
2015)</span>: One of the first widely-used deconvolution tools, uses
support vector regression (requires downloading the CIBERSORT.R script
from the CIBERSORT website)</li>
<li>
<strong>CIBERSORTx</strong> <span class="citation">(Newman et al.
2019)</span>: An enhanced version of CIBERSORT that includes batch
correction features (requires a token and Docker)</li>
<li>
<strong>BayesDeBulk</strong> <span class="citation">(Petralia et al.
2023)</span>: A Bayesian approach specifically designed for proteomics
deconvolution</li>
</ul>
<p>To use a different algorithm, simply change the
<code>algorithm</code> parameter:</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># Example using a different algorithm</span></span>
<span><span class="va">results_cibersort</span> <span class="op">&lt;-</span> <span class="fu"><a href="../reference/deconvolute.html">deconvolute</a></span><span class="op">(</span></span>
<span>  algorithm <span class="op">=</span> <span class="st">"cibersort"</span>,</span>
<span>  data <span class="op">=</span> <span class="va">mixed_samples_preprocessed</span>,</span>
<span>  signature <span class="op">=</span> <span class="va">signature_matrix</span></span>
<span><span class="op">)</span></span></code></pre></div>
</div>
</div>
<div class="section level2">
<h2 id="formatting-and-visualizing-results">Formatting and visualizing results<a class="anchor" aria-label="anchor" href="#formatting-and-visualizing-results"></a>
</h2>
<p>For easier visualization, let’s convert the results to a tidy
format:</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">results_tidy</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://tibble.tidyverse.org/reference/as_tibble.html" class="external-link">as_tibble</a></span><span class="op">(</span><span class="va">results</span>, rownames <span class="op">=</span> <span class="st">"sample"</span><span class="op">)</span> <span class="op">|&gt;</span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="op">-</span><span class="va">sample</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">"cell_type"</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">"cell_fraction"</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="va">results_tidy</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 6 × 3</span></span></span>
<span><span class="co">#&gt;   sample cell_type    cell_fraction</span></span>
<span><span class="co">#&gt;   <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>  <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>                <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">1</span> Mix_1  CD8+ T cells         0.269</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">2</span> Mix_1  Monocytes            0.731</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">3</span> Mix_2  CD8+ T cells         0.334</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">4</span> Mix_2  Monocytes            0.666</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">5</span> Mix_3  CD8+ T cells         0.207</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">6</span> Mix_3  Monocytes            0.793</span></span></code></pre></div>
<p>Now, let’s create a bar chart to visualize the cell type
proportions:</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="va">results_tidy</span>, <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">sample</span>, y <span class="op">=</span> <span class="va">cell_fraction</span>, fill <span class="op">=</span> <span class="va">cell_type</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_bar.html" class="external-link">geom_bar</a></span><span class="op">(</span>stat <span class="op">=</span> <span class="st">"identity"</span>, position <span class="op">=</span> <span class="st">"stack"</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_minimal</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/theme.html" class="external-link">theme</a></span><span class="op">(</span>axis.text.x <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/element.html" class="external-link">element_text</a></span><span class="op">(</span>angle <span class="op">=</span> <span class="fl">45</span>, hjust <span class="op">=</span> <span class="fl">1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"Sample"</span>,</span>
<span>    y <span class="op">=</span> <span class="st">"Estimated Proportion"</span>,</span>
<span>    fill <span class="op">=</span> <span class="st">"Cell Type"</span></span>
<span>  <span class="op">)</span></span></code></pre></div>
<p><img src="proteoDeconv_files/figure-html/unnamed-chunk-13-1.png" width="672"></p>
</div>
<div class="section level2">
<h2 id="conclusion">Conclusion<a class="anchor" aria-label="anchor" href="#conclusion"></a>
</h2>
<p>This vignette demonstrated the basic workflow for proteomics
deconvolution using the <code>proteoDeconv</code> package:</p>
<ol style="list-style-type: decimal">
<li>Load and preprocess your data</li>
<li>Create a cell type signature matrix</li>
<li>Perform deconvolution on mixed samples</li>
<li>Visualize and interpret the results</li>
</ol>
<p>For more advanced usage, refer to the package documentation and <a href="https://computationalproteomics.github.io/proteoDeconv/reference/index.html">function
reference</a>.</p>
</div>
<div class="section level2 unnumbered">
<h2 class="unnumbered" id="references">References<a class="anchor" aria-label="anchor" href="#references"></a>
</h2>
<div id="refs" class="references csl-bib-body hanging-indent" entry-spacing="0">
<div id="ref-newmanRobustEnumerationCell2015" class="csl-entry">
Newman, Aaron M., Chih Long Liu, Michael R. Green, Andrew J. Gentles,
Weiguo Feng, Yue Xu, Chuong D. Hoang, Maximilian Diehn, and Ash A.
Alizadeh. 2015. <span>“Robust Enumeration of Cell Subsets from Tissue
Expression Profiles.”</span> <em>Nature Methods</em> 12 (5): 453–57. <a href="https://doi.org/10.1038/nmeth.3337" class="external-link">https://doi.org/10.1038/nmeth.3337</a>.
</div>
<div id="ref-newmanDeterminingCellType2019" class="csl-entry">
Newman, Aaron M., Chloé B. Steen, Chih Long Liu, Andrew J. Gentles,
Aadel A. Chaudhuri, Florian Scherer, Michael S. Khodadoust, et al. 2019.
<span>“Determining Cell Type Abundance and Expression from Bulk Tissues
with Digital Cytometry.”</span> <em>Nature Biotechnology</em> 37 (7):
773–82. <a href="https://doi.org/10.1038/s41587-019-0114-2" class="external-link">https://doi.org/10.1038/s41587-019-0114-2</a>.
</div>
<div id="ref-petraliaBayesDeBulkFlexibleBayesian2023" class="csl-entry">
Petralia, Francesca, Azra Krek, Anna P. Calinawan, Daniel Charytonowicz,
Robert Sebra, Song Feng, Sara Gosline, et al. 2023.
<span>“<span>BayesDeBulk</span>: <span>A Flexible Bayesian
Algorithm</span> for the <span>Deconvolution</span> of <span>Bulk Tumor
Data</span>.”</span> bioRxiv. <a href="https://doi.org/10.1101/2021.06.25.449763" class="external-link">https://doi.org/10.1101/2021.06.25.449763</a>.
</div>
<div id="ref-racleSimultaneousEnumerationCancer2017" class="csl-entry">
Racle, Julien, Kaat de Jonge, Petra Baumgaertner, Daniel E Speiser, and
David Gfeller. 2017. <span>“Simultaneous Enumeration of Cancer and
Immune Cell Types from Bulk Tumor Gene Expression Data.”</span> Edited
by Alfonso Valencia. <em>eLife</em> 6 (November): e26476. <a href="https://doi.org/10.7554/eLife.26476" class="external-link">https://doi.org/10.7554/eLife.26476</a>.
</div>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Måns Zamore.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.3.</p>
</div>

    </footer>
</div>





  </body>
</html>
